codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -382,43 +403,23 @@ private static void validateHeaderNameElement(byte value) {
         case ':':
         case ';':
         case '=':
-            throw new IllegalArgumentException(
-               "a header name cannot contain the following prohibited characters: =,;: \\t\\r\\n\\v\\f: " +
-                       value);
+            return false;
         default:
-            // Check to see if the character is not an ASCII character, or invalid
-            if (value < 0) {
-                throw new IllegalArgumentException("a header name cannot contain non-ASCII character: " + value);
-            }
+            return true;
         }
     }
 
     private static void validateHeaderNameElement(char value) {
-        switch (value) {
-        case 0x1c:
-        case 0x1d:
-        case 0x1e:
-        case 0x1f:
-        case 0x00:
-        case '\t':
-        case '\n':
-        case 0x0b:
-        case '\f':
-        case '\r':
-        case ' ':
-        case ',':
-        case ':':
-        case ';':
-        case '=':
+        // Check to see if the character is not an ASCII character, or invalid
+        if (value > Byte.MAX_VALUE || value < 0) {
+            throw new IllegalArgumentException("a header name cannot contain non-ASCII character: " +
+                                               value);
+        }
+        final byte byteValue = (byte) value;
+        if (!VALID_HEADER_NAME_ELEMENT[byteValue]) {
```

Comment:
Same comment as above... Basically I think we can just remove the first if check and re throw if `IndexOutOfBoundsException` is throws. This should remove extra branching for the common path.